Makefile

```diff
@@ -4,9 +4,11 @@ deps:
 	mix deps.get
 
 testing: deps
-	mix credo
+	mix do compile --warnings-as-errors --force
 	mix format --check-formatted
+	mix credo
```

Comment:
Could we update this to `credo --strict` please and thank you 😁

Reply:
Seems like Credo is upset about nested modules in about 10 spots. I can ignore that warning in those places, I can fix, or we can do that in a separate PR. Up to you.

Reply:
@doomspork I would prefer to review the `credo --strict` rules fixed in a separate PR.

@asummers probably better to finish with this PR and then make another PR with `--strict` rule turn on and probably will require the fixes to keep the CI happy.

Are you OK with that?

Reply:
Fine by me. I'll rebase this and get it up to date with master then I think this should be good to go!